test/integration/node-specific/resource_tracking_script_builder.ts

```diff
@@ -9,34 +10,52 @@ import { parseSnapshot } from 'v8-heapsnapshot';
 import { type MongoClient } from '../../mongodb';
 import { type TestConfiguration } from '../../tools/runner/config';
 
-export type ResourceTestFunction = (options: {
+export type ResourceTestFunction = HeapResourceTestFunction | ProcessResourceTestFunction;
+
+export type HeapResourceTestFunction = (options: {
   MongoClient: typeof MongoClient;
   uri: string;
   iteration: number;
 }) => Promise<void>;
 
-const RESOURCE_SCRIPT_PATH = path.resolve(__dirname, '../../tools/fixtures/resource_script.in.js');
+export type ProcessResourceTestFunction = (options: {
+  MongoClient: typeof MongoClient;
+  uri: string;
```

Comment:
```suggestion
  uri: string;
  log: (out: any) => void;
```

Then typescript won't be upset in the test files about a log function that doesn't exist in the clost.test.ts file.